jps-plugin/src/org/intellij/erlang/jps/builder/ErlangPrepareDependenciesBuilder.java

```diff
@@ -0,0 +1,283 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+
+public class ErlangPrepareDependenciesBuilder extends TargetBuilder<ErlangSourceRootDescriptor, ErlangPrepareDependenciesTarget> {
+  public static final String DEPENDENCIES_TREE_FILE_PATH = "erlang-builder/deps-tree.xml";
+  private static final String NAME = "Prepare dependences";
+  private static final Logger LOG = Logger.getInstance(ErlangPrepareDependenciesBuilder.class);
+
+
+  public ErlangPrepareDependenciesBuilder() {
+    super(Collections.singletonList(ErlangPrepareDependenciesTargetType.INSTANCE));
+  }
+
+  @Override
+  public void build(@NotNull ErlangPrepareDependenciesTarget target,
+                    @NotNull DirtyFilesHolder<ErlangSourceRootDescriptor, ErlangPrepareDependenciesTarget> holder,
+                    @NotNull BuildOutputConsumer outputConsumer,
+                    @NotNull CompileContext context) throws ProjectBuildException, IOException {
+    LOG.info("Starting prepare dependencies");
+    LOG.debug("Load project build order.");
+    ErlangProjectBuildOrder projectBuildOrder = loadProjectBuildOrder(context);
+    if (projectBuildOrder == null) {
+      addPrepareDependenciesFailedMessage(context);
+      return;
+    }
+
+    LOG.debug("Collect dirty files.");
+    List<String> dirtyErlangFilePaths = new DirtyFileProcessor<String, ErlangPrepareDependenciesTarget>() {
+      @Nullable
+      @Override
+      protected String getDirtyElement(@NotNull ErlangPrepareDependenciesTarget target,
+                                       @NotNull File file,
+                                       @NotNull ErlangSourceRootDescriptor root) throws IOException {
+        return isSourceOrHeader(file) ? file.getAbsolutePath() : null;
+      }
+    }.collectDirtyElements(holder);
+
+    JpsProject project = target.getProject();
+    Map<String, ErlangModuleBuildOrder> moduleBuildOrders;
+    if (dirtyErlangFilePaths.isEmpty()) {
+      LOG.debug("There aren't dirty .erl or .hrl files.");
+      moduleBuildOrders = ContainerUtil.newHashMap();
+    }
+    else {
+      LOG.debug("Search dependent dirty files.");
+      moduleBuildOrders = prepareBuildOrderForDirtyFiles(project, projectBuildOrder, dirtyErlangFilePaths);
+    }
+    LOG.debug("Start writing build orders to XML.");
+    writeBuildOrderToXml(project, context, moduleBuildOrders);
+  }
+
+  @NotNull
+  @Override
+  public String getPresentableName() {
+    return NAME;
+  }
+
+  @Nullable
+  private static ErlangProjectBuildOrder loadProjectBuildOrder(@NotNull CompileContext context) {
+    try {
+
+      return ErlangBuilderXmlUtil.readFromXML(DEPENDENCIES_TREE_FILE_PATH, context, ErlangProjectBuildOrder.class);
+    }
+    catch (JDOMException e) {
+      LOG.warn("Can't read XML from " + DEPENDENCIES_TREE_FILE_PATH);
+      LOG.debug(e);
+    }
+    catch (IOException e) {
+      LOG.warn("I/O exceptions is occurred while reading " + DEPENDENCIES_TREE_FILE_PATH);
+      LOG.debug(e);
```

Comment:
Same stuff with exception severity as before. Also, why not move exception handling logic to the utility method?